### PR TITLE
fix(numpy): return NaN for nanquantile on empty reduction axes

### LINKED ISSUE #39468: jnp.[nan]quantile & jnp.[nan]percentile fails on empty reduction axes
### The Problem

`jnp.nanquantile` raises an indexing or gather error when the
reduction axis is empty, however, NumPy returns `NaNs` with the expected
reduced shape.

### Reproduction

```python
import jax
import jax.numpy as jnp
import numpy as np

x_np = np.empty((0, 4), dtype=np.float32)
x_jax = jnp.asarray(x_np)

print(np.nanquantile(x_np, 0.5, axis=0))
print(jnp.nanquantile(x_jax, 0.5, axis=0))
print(jax.jit(lambda x: jnp.nanquantile(x, 0.5, axis=0))(x_jax))
```
NumPy returns:
```
[nan nan nan nan]
```

JAX instead raises an indexing / gather error.
but JAX should match:
```
np.nanquantile(np.empty((0, 4)), 0.5, axis=0)
```
and return `NaNs` with the appropriate reduced
shape for an empty reduction axis.

I also observed the failure with:
- `axis=None`
- tuple-valued axes
- ` keepdims=True`
- scalar and vector `q`
- eager execution and `jax.jit`

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.11.1.dev20260728+7238390c24
jaxlib: 0.11.0
numpy:  2.5.1
python: 3.12.12 (main, Oct 14 2025, 21:38:21) [Clang 20.1.4 ]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Darwin', node='Eds-MacBook-Pro.local', release='24.6.0', version='Darwin Kernel Version 24.6.0: Wed Nov  5 21:30:44 PST 2025; root:xnu-11417.140.69.705.2~1/RELEASE_ARM64_T6041', machine='arm64')
```

### PR DESCRIPTION
**Pull request by AloysJehwin**
## Description

Fixes #39468

`jnp.nanquantile` raised an indexing/gather error when the reduction axis had size 0:

```python
x = jnp.empty((0, 4), dtype=jnp.float32)
jnp.nanquantile(x, 0.5, axis=0)  # raised IndexError
```

NumPy's `nanquantile` falls back to `nanmean` for empty axes, returning NaN with the reduced shape. The fix adds an early return in `_quantile()` that delegates to `nanmean` when `squash_nans=True` and the reduction axis is empty.

## Testing

Added `testNanquantileEmptyAxis` in `tests/lax_numpy_reducers_test.py` covering:
- scalar `q`, `axis=0` on shape `(0, 4)` → shape `(4,)` of NaNs
- `axis=1` on shape `(4, 0)` → shape `(4,)` of NaNs
- `keepdims=True` → shape `(1, 4)` of NaNs

Note: full test suite requires nightly `jaxlib` (>= 0.11.0) not yet on PyPI. The fix was validated against NumPy's output shapes for all tested cases.

Implemented with Claude Code assistance. All changed lines reviewed manually.

### CODE REVIEW

**Comment by AloysJehwin** [comment]
Closing in favour of #39475 which was opened first. Thanks for working on this!
